### ISSUE TITLE
Add static var to get all plus devices

### ISSUE THREAD

**Issue by lucasecf** [dennisweissmann/DeviceKit#101]
Talking about phones, The plus versions can have some different spacing or behaviours than "regular" sized devices (4.7inches and below).

In our project, we have some complex custom designs and currently we have this line of code there:
```
Device().isOneOf([.iPhone6Plus, .iPhone6sPlus, .iPhone7Plus, .iPhone8Plus])
```

Which is not ideal, because we have to manually update it everytime we have new iPhones released.
Instead, it would be nice to have something like:
```
Device().isOneOf(Device.allPlusPhones)
```

What do you think? If you agree, I can create a small PR adding the method + a small test

**Comment by dennisweissmann** [comment]
Yeah that'd be super cool :) I wanted to do the same for the "Pro" iPads .. Feel free to go ahead :)

I don't know what the best approach would be though, maybe add another column (i.e. property) in the python devices array or just manually put them into a method 🤔

The advantage of the former is that we don't forget to update the method when new devices are added (which is what happened before), this is also the disadvantage of the latter approach ...
The disadvantage of the former is that every device in the python array would have an added property, which is unnecessary in most cases ...

I don't have a strong opinion on that, but I think I slightly prefer the first approach (adding a `isPlusFormFactor` and `isProDevice` or something along those lines to the overall array).

What do you think? What do others think?

OTOH the complete generation thingy is an implementation detail which change anytime without breaking the API so it's not the worst thing in the world if we make a "wrong" decision but it would be great to have a concept for things like that.

**Comment by lucasecf** [comment]
@dennisweissmann I was thinking in a rather simple approach - get allPhones array and filter the ones in which the name contains "Plus" or similar... 

This also wouldn't need the manual maintenance and wouldn't touch the Devices structure. If the naming pattern ever change, by definition a `allPlusPhones` var/method would also be deprecated/invalid anyhow.

Just not sure how to do that in the whole python generation setup stuff, never used it before, but I guess I can figure it out.

If you still think the `isPlusFormFactor` flag is actually better/safer, I can go this way also, don't mind that much

**Comment by dennisweissmann** [comment]
Hm, your approach can become quite runtime-intensive. I'd actually prefer the compile-time approach I think.

Let me know if you wanna play around with it, otherwise I'll try to block some time this weekend and do it :)

**Comment by lucasecf** [comment]
Don't get very much the runtime argument. Isn't the runtime check happening in the python code, during the code generations phase? If so, why this matters if everyone will always use only the outputted swift code, containing the "fixed" generated devices?

Anyhow, I would try to find some time only next week, not in the weekend, so even better if you can find some time to do this before me 😄 👍

**Comment by dennisweissmann** [comment]
Maybe I misunderstood you :)

I thought you wanna do (the following are the generated swift files):

    var allPlusPhones: [Device] {
      return allPhones.filter { $0.description.contains("Plus") }
    }

which is (IMO) not the nicest and future-proof way to test for the plus form factor.

My approach would (in the generated swift file) look like your code sample from above:

    static let allPlusPhones: [Device] = [.iPhone6Plus, .iPhone6sPlus, .iPhone7Plus, .iPhone8Plus]

I'll try to get to it later :)

**Comment by dennisweissmann** [comment]
This is fixed in 1.4.0, #102 

Thanks for your suggestion :)

**Comment by lucasecf** [comment]
awesome, thanks very much 💯 👍
